code/modules/mob/living/simple_animal/guardian/guardian.dm

```diff
@@ -141,6 +141,11 @@ GLOBAL_LIST_EMPTY(parasites) //all currently existing/living guardians
 			summoner.dust()
 			death(TRUE)
 			qdel(src)
+		if(QDELETED(summoner))
```

Comment:
just change the above if(summoner) to if(!QDELETED(summoner))

Reply:
okay